synapse/tests/test_lib_crypto_tinfoil.py

```diff
@@ -0,0 +1,96 @@
+import hashlib
+import binascii
+
+from cryptography.hazmat.backends import default_backend
+
+import synapse.lib.crypto.tinfoil as s_tinfoil
+
+from synapse.tests.common import *
+
+class TinFoilTest(SynTest):
+
+    def test_lib_crypto_tnfl_base(self):
+
+        ekey = s_tinfoil.newkey()
+        self.len(32, ekey)
+        self.isinstance(ekey, bytes)
+
+        # Keys are random from s_tinfoil.newkey
+        self.ne(ekey, s_tinfoil.newkey())
+        self.ne(ekey, s_tinfoil.newkey())
+
+        tinh = s_tinfoil.TinFoilHat(ekey)
+        self.true(tinh.bend is default_backend())
+
+        byts = tinh.enc(b'foobar')
+
+        # Ensure the envelope is shaped as we expect it too be
+        edict = s_msgpack.un(byts)
+        self.isinstance(edict, dict)
+        self.len(3, edict)
+
+        data = edict.get('data')
+        self.isinstance(data, bytes)
+        self.gt(len(data), 1)
+
+        iv = edict.get('iv')
+        self.isinstance(iv, bytes)
+        self.gt(len(iv), 1)
+
+        hmac = edict.get('hmac')
+        self.isinstance(hmac, bytes)
+        self.gt(len(hmac), 1)
```

Comment:
Shouldn't these lengths always be 16, 16, 32 (respectively)?

Reply:
iv and hmac lengths are known; and data should be in multiples of the block size due to padding. can update tests accordingly.

Reply:
added

Reply:
ya that's an even better idea